fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingMultiTargetByRecords.java

```diff
@@ -172,12 +170,13 @@ private CompletableFuture<Boolean> buildRangeOnly(@Nonnull FDBRecordStore store,
             if (range == null) {
                 return AsyncUtil.READY_FALSE; // no more missing ranges - all done
             }
-            final Tuple rangeStart = RangeSet.isFirstKey(range.begin) ? null : Tuple.fromBytes(range.begin);
-            final Tuple rangeEnd = RangeSet.isFinalKey(range.end) ? null : Tuple.fromBytes(range.end);
-            final TupleRange tupleRange = TupleRange.between(rangeStart, rangeEnd);
+            final byte[] keyPrefix = store.recordsSubspace().pack();
+            final byte[] rangeStart = ByteArrayUtil.join(keyPrefix, range.begin);
+            final byte[] rangeEnd = ByteArrayUtil.join(keyPrefix, range.end);
+            final KeyRange keyRange = new KeyRange(rangeStart, rangeEnd);
```

Comment:
I'm trying to wrap my head around the `KeyRange`. Looking at `TupleRange::toRange` (which is still used to translate a `KeyRange` into an FDB `Range`), a `KeyRange` where the low endpoint is `RANGE_INCLUSIVE` and the end endpoint is always `RANGE_EXCLUSIVE` will always behave just like a `Range` over those same two keys. But it also behaves just like a `TupleRange` with packed elements, so for instance, something like `new KeyRange(bytes, EndpointType.RANGE_INCLUSIVE, bytes, EndpointType.RANGE_INCLUSIVE)` represents all keys "strictly" prefixed by the range (that is, excluding keys prefixed by `bytes + '\xff'`).

That feels fair enough.